src/avram/save_operation.cr

```diff
@@ -339,6 +339,37 @@ abstract class Avram::SaveOperation(T)
     !!record_id
   end
 
+  # `#persisted?` always returns `true` in `after_*` hooks, whether
+  # a new record was created, or an existing one was updated.
+  #
+  # This method should always return `true` for a create or `false`
+  # for an update, independent of the stage we are at in the operation.
+  def new_record? : Bool
+    {{ T.resolve.constant(:PRIMARY_KEY_NAME).id }}.value.nil?
+  end
+
+  def revert : self?
+    return unless saved?
+
+    saved_record = record.not_nil!
+    operation = self.class.new(saved_record)
+
+    if new_record?
+      operation if saved_record.delete.rows_affected > 0
```

Comment:
What happens if the `saved_record` is SoftDeletable? Also, all record deleting should start moving in to the new `DeleteOperation`.

Reply:
> What happens if the `saved_record` is SoftDeletable?

It is hard-deleted, since we're reverting the creation of that record. I'm not very familiar with soft deletes, though.

> Also, all record deleting should start moving in to the new `DeleteOperation`.

Not familiar with `DeleteOperation` either. I'll look into it, but I may need help.